spec/preservation_ingest/update_catalog_spec.rb

```diff
@@ -0,0 +1,55 @@
+describe Robots::SdrRepo::PreservationIngest::UpdateCatalog do
+  subject(:update_catalog_obj) { described_class.new }
+
+  let(:bare_druid) { 'bj102hs9687' }
+  let(:size) { 2342 }
+  let(:version) { 3 }
+  let(:strg_root) { "some/storage/location/from/endpoint/table" }
+  let(:faraday_dbl) { class_double(Faraday) }
+  let(:url) { 'http://localhost:3000' }
+  let(:args) do
+    { druid: bare_druid, incoming_version: version, incoming_size: size, storage_location: strg_root }
+  end
+  let(:mock_so) do
+    instance_double(Moab::StorageObject, object_pathname: instance_double(Pathname), size: size, storage_root: strg_root, current_version_id: version)
+  end
+
+  describe '#perform' do
+    before do
+      allow(Moab::StorageServices).to receive(:find_storage_object).and_return(mock_so)
+      allow(LyberCore::Log).to receive(:debug).with(any_args)
+      # allow(Faraday).to receive(:new).with(url: url).and_return(faraday_dbl)
+
+    end
+
+    context 'object is new' do
+      it 'POST to /catalog' do
+        response = instance_double(Faraday::Response, status: 201)
+        allow(Faraday).to receive(:new).with(url: url).and_return(faraday_dbl)
+        expect(faraday_dbl).to receive(:post).with('/catalog', args).and_return(response)
+        expect(faraday_dbl).not_to receive(:patch)
+        update_catalog_obj.perform(bare_druid)
+      end
+    end
+
+    context 'object already exists' do
+      it 'PATCH to /catalog/:druid' do
+        response = instance_double(Faraday::Response, status: 409)
+        allow(Faraday).to receive(:new).with(url: url).and_return(faraday_dbl)
+        expect(faraday_dbl).to receive(:post).with('/catalog', args).and_return(response)
+        expect(faraday_dbl).to receive(:patch).with("/catalog/#{bare_druid}", args).and_return(response)
+        update_catalog_obj.perform(bare_druid)
+      end
+    end
+
+    context 'ensure connection calls the right Setting' do
```

Comment:
Does it make sense to bust this out to not be about `#perform` at all, but about `#conn`?  I think the main point here isn't related to `perform`, but to the the fact `conn` draws from the right spot in `Settings`.

Reply:
good point. will make the change